core/src/main/java/hudson/model/RunMap.java

```diff
@@ -231,6 +231,8 @@ protected R retrieve(File d) throws IOException {
                 LOGGER.log(Level.WARNING, "could not load " + d, e);
             } catch (InstantiationError e) {
                 LOGGER.log(Level.WARNING, "could not load " + d, e);
+            } catch (Exception e) {
```

Comment:
Generally yes, catching `Throwable` is fine too. (If the application ever makes use of `ThreadDeath` you would want to rethrow that, though Jenkins does not AFAIK.)